sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -365,7 +366,7 @@ def _create_handler(self, auth: Union["pyamqp_JWTTokenAuth", "uamqp_JWTTokenAuth
             if self._max_wait_time
             else 0,
             # set link_credit to at least 1 so that messages can be received
-            link_credit=self._prefetch_count + 1,
+            link_credit=self._prefetch_count + 1 if self._prefetch_count else self._message_count,
```

Comment:
if max_message_count is set on recieve_messages, we were not asking the service for the correct amount of link_credits, we were simply flowing a link credit of 1 after our initial flow frame of link credits were passed along